website/addons/github/model.py

```diff
@@ -76,10 +76,7 @@ def revoke_remote_oauth_access(self, external_account):
     # Assumes oldest connected account is primary.
     @property
     def public_id(self):
-        gh_accounts = [
-            a for a in self.owner.external_accounts.all()
-            if a.provider == self.oauth_provider.short_name
-        ]
+        gh_accounts = self.owner.external_accounts.filter(provider=self.oauth_provider.short_name)
         if gh_accounts:
             return gh_accounts[0].display_name
```

Comment:
You probably want `.exists()` which will return a bool. `.first()` will cause a sort, which depending on the table and its indexes could be really really slow.